File: code/java/strStr.java
class Solution {
    public int strStr(String haystack, String needle) {
        if(needle.length() > haystack.length())
            return -1;
        else if(needle.length() == haystack.length())
            if (needle.equals(haystack) == true)
                return 0;
            else
                return -1;
        else{
            String temp;
            for(int i=0;i+needle.length()<=haystack.length();i++){
                temp = haystack.substring(i,i+needle.length());
                if (needle.equals(temp) == true)
                    return i;
            }
            return -1;
        }
    }
}

File: code/python/strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(haystack) < len(needle):
            return -1
        elif len(haystack) == len(needle):
            if haystack == needle:
                return 0
            else:
                return -1
        else:

            for i in range(0,len(haystack)-len(needle)+1):
                if needle == haystack[i:i+len(needle)]:
                    return i
            return -1
